js/data/abstract_store.js

```diff
@@ -309,21 +310,61 @@ var Store = Class.inherit({
     remove: function(key) {
         var that = this;
 
-        that.fireEvent("modifying");
-        that.fireEvent("removing", [key]);
+        that._eventsStrategy.fireEvent("modifying");
+        that._eventsStrategy.fireEvent("removing", [key]);
 
         return that._addFailHandlers(that._removeImpl(key).done(function(callbackKey) {
-            that.fireEvent("removed", [callbackKey]);
-            that.fireEvent("modified");
+            that._eventsStrategy.fireEvent("removed", [callbackKey]);
+            that._eventsStrategy.fireEvent("modified");
         }));
     },
 
     _removeImpl: abstract,
 
     _addFailHandlers: function(deferred) {
         return deferred.fail(this._errorHandler).fail(errorsModule._errorHandler);
+    },
+
+    /**
+     * @name StoreMethods.on
+     * @publicName on(eventName, eventHandler)
+     * @param1 eventName:string
+     * @param2 eventHandler:function
+     * @return this
+     */
+    /**
+     * @name StoreMethods.on
+     * @publicName on(events)
+     * @param1 events:object
+     * @return this
+     */
+    on: function(eventName, eventHandler) {
+        if(isPlainObject(eventName)) {
+            each(eventName, (function(e, h) { this.on(e, h); }).bind(this));
+        } else {
+            this._eventsStrategy.on(eventName, eventHandler);
+        }
+        return this;
+    },
+
+    /**
+     * @name StoreMethods.off
+     * @publicName off(eventName)
+     * @param1 eventName:string
+     * @return this
+     */
+    /**
+     * @name StoreMethods.off
+     * @publicName off(eventName, eventHandler)
+     * @param1 eventName:string
+     * @param2 eventHandler:function
+     * @return this
+     */
+    off: function(eventName, eventHandler) {
+        this._eventsStrategy.off(eventName, eventHandler);
+        return this;
```

Comment:
Seems like we should move these methods into EventsStrategy.

And use here something like that

```js
off: function(eventName, eventHandler) {
  this._eventsStrategy.offMethod2(eventName, eventHandler);
  return this;
}
```